### PR TITLE
p_x_interval

### PR DESCRIPTION
**Pull request by wcbeard**
Implemented another equation from the paper that gives probability of returning x times over interval of n days.

### CODE REVIEW

File: bgbb/core.py [hunk 1]
```diff
@@ -136,12 +123,95 @@ def _expec_p_th(cls, a, b, g, d, f, r, n):
...
+        Implements the probability that a customer with purchase history (x, tx, n)
+        makes x* transactions in the interval (n, n + n*],
+        AKA Equation (11) from "Customer-Base Analysis in a Discrete-Time
+        Noncontractual Setting," Fader 2009.
+        You can get the probability that a user will *ever* return by computing
+        this quantity for x_star == 0, and then subtracting it from 1.
+        """
+        # Nomenclature of the paper: x, tx, n
+        x, tx = f, r
+        del r, f
```
Comment (acmiyaguchi) [inline bgbb/core.py]: It's unnecessary to remove variables from the local namespace.

File: tests/test_bgbb.py [hunk 2]
```diff
@@ -69,15 +76,48 @@ def test_expec_p_th(bg):
...
+
+    assert check_prob(p_alive)
+    assert check_prob(p_never_return)
+    assert check_prob(p_ever_return)
+
+    assert np.corrcoef(p_alive, p_ever_return)[0, 1] > .99
+
+    # Test case for x_star != 0
+    p_return_once = bgdat.rfn.p_x_interval(n_star=100, x_star=1)
+    assert check_prob(p_return_once)
```
Comment (acmiyaguchi) [inline tests/test_bgbb.py:111]: Is `p_return_once` also close to `p_alive`, as per the name of the test?

**Review by acmiyaguchi — APPROVED** [review]
I double checked your work against equation 11 of the paper -- it looks like everything is captured correctly.

File: tests/test_bgbb.py (hunk 2 above)
Comment (wcbeard) [reply tests/test_bgbb.py:111]: So if they're alive, the probability is high that they'll return between 1 times-n times where n is a large number. For a large interval (say a month) the probability is pretty low that you'll return *exactly* once. But you can get the probability that they'll return any non-zero # times in the interval by taking 1 - P(return exactly 0 times). And I'm assuming this is a good proxy for p_alive. Hopefully make sense.

File: bgbb/core.py (hunk 1 above)
Comment (wcbeard) [reply bgbb/core.py]: Good catch, thanks.

**Pull request merged**